week_03/ykshin/2020907_ykshin_Coin0.kt

```diff
@@ -0,0 +1,29 @@
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.lang.Integer.min
+import java.lang.System.`in`
+import java.util.*
+
+fun main() {
+    val bufferedReader = BufferedReader(InputStreamReader(`in`))
+    var stringTokenizer = StringTokenizer(bufferedReader.readLine())
+    val n = stringTokenizer.nextToken().toInt()
+    val k = stringTokenizer.nextToken().toInt()
+    val coins = IntArray(n)
+    for(i in 0 until n){
+        coins[i] = StringTokenizer(bufferedReader.readLine()).nextToken().toInt()
+    }
+
+    var sum = k
+    var count = 0
+    for(i in coins.size-1 downTo 0){
+        val no = sum / coins[i]
+        count+=no
+        sum-=no * coins[i]
+    }
```

Comment:
```suggestion
    var sum_of_money = k
    var count = 0
    for(index in coins.size - 1 downTo 0){
        val quotient = sum_of_money / coins[index]
        count += quotient
        sum_of_money -= quotient * coins[index]
    }
```

그냥 나였으면 저렇게 써줬을 거 같애ㅋㅋ

Reply:
그리고 qutotient 가 0일 때는 아래 연산들이 의미 없이 수행되는 거 같은데,

저 for문에 조건문을 넣어서 의미 없는 연산들은 굳이 안하도록 하는 것은 어떨까~? :smile:
ex) quotient 가 0 일 때 / quotient가 0보다 클 때

Reply:
작은 변수여도 조금더 정확한 의미를 갖도록 구성하겠습니다! 감사합니다!